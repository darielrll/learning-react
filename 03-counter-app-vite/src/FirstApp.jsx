// import { Fragment } from "react";
import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {

    // console.log(props);
    return (
        <>
            <h1 data-testid="test-title"> { title } </h1>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{name}</p>
        </>            
    );
}

FirstApp.propTypes = {  
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Dariel',
    subTitle: 'no hay subt√≠tulo',
    // title: 'no hay titulo',
}