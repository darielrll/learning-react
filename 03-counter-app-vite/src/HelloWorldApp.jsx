
export const HelloWorldApp = () => {
    return (
        <div>Hola Mundo App</div>
    );
}