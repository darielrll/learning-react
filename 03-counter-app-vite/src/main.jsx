import React from 'react'
import ReactDOM from 'react-dom/client'
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <HelloWorldApp/> */}
        {/* <FirstApp title="Hola, soy Vegueta"/> */}
        <CounterApp value={3}/>
    </React.StrictMode>
    );