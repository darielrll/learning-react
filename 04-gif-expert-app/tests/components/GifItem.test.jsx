
import { render, screen, fireEvent } from "@testing-library/react"
import { GifItem } from "../../src/components";

describe('Pruebas a GifItem', () => { 

    const title = "Test Title";
    const url = "https://testurl.com/test.gif";

    test('debe hacer match con el snapshot', () => {
        const {container} = render( <GifItem  title={title} url={url}/> )       

        expect(container).toMatchSnapshot()
    })

    // test('debe mostrar la url y el title', () => {
    //     const title = "Test Title";
    //     const url = "https://testurl.com/test.gif";
    //     render( <GifItem title={title} url={url} /> )       

    //     const imgElement = screen.getByRole('img');
    //     expect(imgElement.src).toBe(url);
    //     expect(imgElement.alt).toBe(title);

    //     expect(screen.getByText(title)).toBeTruthy();
    // })

    // test('debe lanzar error si no se reciben title y url', () => {
        
    //     render( <GifItem  /> )

    //     // console.error = jest.fn(); // Mock console.error to suppress error output in test

    //     // expect(() => render( <GifItem /> )).toThrow();
    //     // expect(console.error).toHaveBeenCalled();
    // })

})