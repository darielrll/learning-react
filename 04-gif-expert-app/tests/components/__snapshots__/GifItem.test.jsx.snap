// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="https://testurl.com/test.gif"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
